src/main/java/org/sefglobal/scholarx/service/CommentService.java

```diff
@@ -0,0 +1,102 @@
+package org.sefglobal.scholarx.service;
+
+import org.sefglobal.scholarx.exception.BadRequestException;
+import org.sefglobal.scholarx.exception.ResourceNotFoundException;
+import org.sefglobal.scholarx.exception.UnauthorizedException;
+import org.sefglobal.scholarx.model.Comment;
+import org.sefglobal.scholarx.model.Mentee;
+import org.sefglobal.scholarx.model.Profile;
+import org.sefglobal.scholarx.repository.CommentRepository;
+import org.sefglobal.scholarx.repository.MenteeRepository;
+import org.sefglobal.scholarx.repository.ProfileRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CommentService {
+    private final static Logger log = LoggerFactory.getLogger(CommentService.class);
+    private final CommentRepository commentRepository;
+    private final ProfileRepository profileRepository;
+    private final MenteeRepository menteeRepository;
+
+    public CommentService(ProfileRepository profileRepository,MenteeRepository menteeRepository,
+                                CommentRepository commentRepository){
+        this.commentRepository = commentRepository;
+        this.profileRepository = profileRepository;
+        this.menteeRepository = menteeRepository;
+    }
+
+    public List<Comment> getAllMenteeComments(long menteeId, long profileId)
+            throws ResourceNotFoundException, BadRequestException, UnauthorizedException {
+        Optional<Mentee> optionalMentee = menteeRepository.findById(menteeId);
+
+        if (!optionalMentee.isPresent()) {
+            String msg = "Error, Mentee by id: " + menteeId + " doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        } else if (optionalMentee.get().getProfile().getId() == profileId) {
```

Comment:
Throw unauthorized exception if the profile doesn't belong to the assignedMentor or an admin.

Reply:
ok

Reply:
We'll have to use the validation used in line 68-77